server/src/main/java/org/elasticsearch/cluster/ClusterState.java

```diff
@@ -752,7 +766,7 @@ public ClusterState build() {
                 uuid = UUIDs.randomBase64UUID();
             }
             return new ClusterState(clusterName, term, version, uuid, metaData, routingTable, nodes, blocks, customs.build(),
-                lastCommittedConfiguration, lastAcceptedConfiguration, fromDiff);
+                lastCommittedConfiguration, lastAcceptedConfiguration, Collections.unmodifiableSet(votingTombstones), fromDiff);
```

Comment:
also create copy of set here? o.w. some can still modify the set using original builder. Alternatively you can create new sets instead of calling `add` and `clear`

Reply:
Thanks, done.